include/GammaEquationSystem.h

```diff
@@ -71,20 +74,30 @@ class GammaEquationSystem : public EquationSystem {
   void reinitialize_linear_system();
   
   void predict_state();
+
+  void normalize_dwalldistdx();
+  void compute_norm_dot_vel();
+
   void assemble_nodal_gradient();
+  void assemble_walldist_gradient();
+  void assemble_ndotv_gradient();
   void compute_effective_diff_flux_coeff();
   
   const bool managePNG_;
 
   ScalarFieldType *gamma_;
   VectorFieldType *dgamdx_;
+  VectorFieldType *dWallDistdx_;
+  VectorFieldType *dNDotVdx_;
   ScalarFieldType *gamTmp_;
   ScalarFieldType *minDistanceToWall_;
+  ScalarFieldType *NDotV_;
   ScalarFieldType *visc_;
   ScalarFieldType *tvisc_;
   ScalarFieldType *evisc_;
   ScalarNodalGradAlgDriver nodalGradAlgDriver_;
-  
+  ScalarWallDistGradAlgDriver walldistGradAlgDriver_;
+  ScalarNDotVGradAlgDriver ndotvGradAlgDriver_;
```

Comment:
```suggestion
  ScalarNodalGradAlgDriver walldistGradAlgDriver_;
  ScalarNodalGradAlgDriver ndotvGradAlgDriver_;
```
I don't think you need two whole new classes if there isn't a new functionality.  You just need new objects to handle the gradient computations.  Please test this out and let me know if it works. With this change you can delete all the `NDotVGrad` and `WallDistGrad` files.

Reply:
@psakievich I have made the requested changes and pushed my mods. Please let me know if there are any outstanding issues to be resolved. Hopefully, we can wrap up the MR soon.